saleor/shipping/models.py

```diff
@@ -165,32 +166,44 @@ def applicable_shipping_methods_for_instance(
         instance: Union["Checkout", "Order"],
         channel_id,
         price: Money,
+        shipping_address: Optional["Address"] = None,
         country_code: Optional[str] = None,
         lines: Union[
             Iterable["CheckoutLineInfo"], Iterable["OrderLineInfo"], None
         ] = None,
     ):
-        if not instance.shipping_address:
+        if not shipping_address:
             return None
+
         if not country_code:
-            # TODO: country_code should come from argument
-            country_code = instance.shipping_address.country.code
+            country_code = shipping_address.country.code
+
         if lines is None:
             # TODO: lines should comes from args in get_valid_shipping_methods_for_order
             lines = list(instance.lines.prefetch_related("variant__product").all())  # type: ignore[misc] # this is hack # noqa: E501
         instance_product_ids = {
             line.variant.product_id for line in lines if line.variant
         }
+
+        from ..checkout.models import Checkout
```

Comment:
Do we need this?

Reply:
It's used in the `isinstance` check below. Also it's local import due to circular dependencies, but fixing them would require too much refactoring for now.